app/src/main/java/com/example/chattutorial/ChannelActivity.kt

```diff
@@ -15,20 +15,17 @@ import com.getstream.sdk.chat.viewmodel.messages.bindView
 import io.getstream.chat.android.client.models.Channel
 import kotlinx.android.synthetic.main.activity_channel.*
 
-/**
- * Show the messages for a channel
- *
- */
 class ChannelActivity : AppCompatActivity(R.layout.activity_channel) {
 
-    private val cid by lazy {
-        intent.getStringExtra(CID_KEY)
+    private val cid: String by lazy {
+        intent.getStringExtra(CID_KEY)!!
     }
 
     override fun onCreate(savedInstanceState: Bundle?) {
         super.onCreate(savedInstanceState)
-
-        val messagesViewModel = MessageListViewModel(cid)
+        val viewModelProvider = ViewModelProvider(this, ChannelViewModelsFactory(cid))
+        //TODO Set ViewHolderFactory
```

Comment:
```suggestion
        // TODO set custom ViewHolderFactory
```

I've reworded this comment a bit in the tutorial, like this